disnake/client.py

```diff
@@ -385,11 +415,11 @@ def __init__(
             VoiceClient.warn_nacl = False
             _log.warning("PyNaCl is not installed, voice will NOT be supported")
 
-        i18n_strict: bool = options.pop("strict_localization", False)
-        i18n = options.pop("localization_provider", None)
-        if i18n is None:
-            i18n = LocalizationStore(strict=i18n_strict)
-        self.i18n: LocalizationProtocol = i18n
+        self.i18n: LocalizationProtocol = (
+            LocalizationStore(strict=strict_localization)
+            if localization_provider is None
+            else localization_provider
+        )
```

Comment:
it might be nice to error if strict_localisation and localization_provider are both provided as strict_localisation doesn't affect the provided localisation provider

Reply:
Seems like a good idea, yeah. Implemented this as part of 1a642234474e2777ce36aa5d754ca488f2cd2024.